src/babel-serializer/file.cc

```diff
@@ -96,3 +96,7 @@ void babel::file::write_lines(cc::string_view filename, cc::range_ref<cc::string
             file.write(line.data(), line.size());
     });
 }
+
+bool babel::file::exists(cc::string_view filename) { return std::ifstream(cc::string(filename).c_str()).good(); }
+
+babel::file::file_output_stream::file_output_stream(cc::string_view filename) { fopen(cc::string(filename).c_str(), "w"); }
```

Comment:
do we want this to be binary mode?

Reply:
ah yeah that'd be better. No random \r in my files please!